neverbleed.c

```diff
@@ -568,23 +778,47 @@ int neverbleed_load_private_key_file(neverbleed_t *nb, SSL_CTX *ctx, const char
 
     if (expbuf_read(&buf, thdata->fd) != 0)
         dief(errno != 0 ? "read error" : "connection closed by daemon");
-    if (expbuf_shift_num(&buf, &ret) != 0 || expbuf_shift_num(&buf, &key_index) != 0 || (estr = expbuf_shift_str(&buf)) == NULL ||
-        (nstr = expbuf_shift_str(&buf)) == NULL || (errstr = expbuf_shift_str(&buf)) == NULL) {
+    if (expbuf_shift_num(&buf, &ret) != 0 || expbuf_shift_num(&buf, &type) != 0 || expbuf_shift_num(&buf, &key_index) != 0) {
         errno = 0;
         dief("failed to parse response");
     }
 
-    if (ret != 1) {
+    switch (type) {
+    case NEVERBLEED_TYPE_RSA:
+        if ((estr = expbuf_shift_str(&buf)) == NULL || (nstr = expbuf_shift_str(&buf)) == NULL) {
+            errno = 0;
+            dief("failed to parse response");
+        }
+        pkey = create_pkey(nb, key_index, estr, nstr);
+        break;
+#if OPENSSL_101_API
+    case NEVERBLEED_TYPE_ECDSA:
```

Comment:
It _might_ be a good idea to create a block within the case (i.e. `case NEVERBLEED_TYPE_ECDSA: { ... } break;`) and declare the variables only for ECDSA-related operation to suppress following warnings:
```
neverbleed.c:770:11: warning: unused variable 'ec_pubkeystr' [-Wunused-variable]
    char *ec_pubkeystr;
          ^
neverbleed.c:771:12: warning: unused variable 'curve_name' [-Wunused-variable]
    size_t curve_name;
           ^
```

And the same thing could be done for RSA as well, I guess.